### PR TITLE
services.synergy: Use graphical target (fixes #9468)

### LINKED ISSUE #9468: Synergy: Autostart option starts regardless of whether X11 is up.
Synergy crashes without an X Display available, but the autostart option attempts to start synergy  regardless of whether login is graphical or TTY. It would probably make more sense to spawn this after graphical.target.

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/synergy.nix#L98

### PR DESCRIPTION
**Pull request by WhittlesJr**
###### Motivation for this change
Synergy client would not auto-start, because it tried to come up before there was a graphical target.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/misc/synergy.nix [hunk 1]
```diff
@@ -83,20 +83,20 @@ in
...
+        wantedBy = optional cfgC.autoStart "graphical-session.target";
         path = [ pkgs.synergy ];
         serviceConfig.ExecStart = ''${pkgs.synergy}/bin/synergyc -f ${optionalString (cfgC.screenName != "") "-n ${cfgC.screenName}"} ${cfgC.serverAddress}'';
         serviceConfig.Restart = "on-failure";
       };
     })
     (mkIf cfgS.enable {
-      systemd.services."synergy-server" = {
-        after = [ "network.target" ];
+      systemd.user.services."synergy-server" = {
```
Comment (Infinisil) [inline nixos/modules/services/misc/synergy.nix]: Does the server really need X11 for it to work?

Also, these services run in the user systemd session with this change, can this change behaviour? Is this intended like this?

Comment (WhittlesJr) [reply nixos/modules/services/misc/synergy.nix]: I believe this is all necessary, per #9468

Comment (Infinisil) [reply nixos/modules/services/misc/synergy.nix]: Alright then :D

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Oh one more thing: Can you change the commit message to use the `nixos/synergy` prefix? I can then merge it

**Comment by Enzime** [comment]
@WhittlesJr can you fix the commit message?

**Comment by WhittlesJr** [comment]
Arg, sorry about that.

**Pull request merged**